File: weather-demo-app-android/app/src/main/java/com/demo/weather/ui/welcome/WelcomeView.kt
package com.demo.weather.ui.welcome

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

@Composable
fun WelcomeView(
    onContinueClicked: () -> Unit,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier
            .fillMaxSize()
            .padding(24.dp),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = "Welcome",
            style = MaterialTheme.typography.displayLarge,
        )

        Spacer(modifier = Modifier.height(12.dp))

        Text(
            text = "Search for a city and view the current weather and forecast.",
            style = MaterialTheme.typography.bodyLarge,
            textAlign = TextAlign.Center
        )

        Spacer(modifier = Modifier.height(24.dp))

        Button(onClick = onContinueClicked) {
            Text("Get Started")
        }
    }
}

File: weather-demo-app-apple/Weather/UI/Welcome/WelcomeView.swift
//
//  WelcomeView.swift
//  Weather
//
//  Created by Onur on 4/11/26.
//

import SwiftUI

struct WelcomeView: View {
    let onContinueClicked: () -> Void

    var body: some View {
        VStack() {
            Text("Welcome")
                .font(.largeTitle)
            
            Spacer().frame(height: 12)

            Text("Search for a city and view the current weather and forecast.")
                .font(.body)
                .multilineTextAlignment(.center)
            
            Spacer().frame(height: 24)

            Button("Get Started", action: onContinueClicked)
                .buttonStyle(.borderedProminent)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .padding(24)
    }
}
